src/worker/tasks/chainIndexer.ts

```diff
@@ -42,6 +43,22 @@ export const createChainIndexerTask = async (args: {
             return;
           }
 
+          // Ensuring that the block data exists.
+          // Sometimes the RPC providers nodes are aware of the latest block
+          // but the block data is not available yet.
+          const block = await provider.getBlockWithTransactions(toBlock);
+
+          if (!block) {
+            logger({
+              service: "worker",
+              level: "warn",
+              message: `Block data not available: ${toBlock} on chain: ${chainId}, url: ${
+                (provider as StaticJsonRpcBatchProvider).connection.url
```

Comment:
We're finding ourselves doing this a few times now. Should prob just have a helper func `getProviderUrl(provider)`